### PR TITLE
Task/DES-2707: add script to migrate project systems to v3 tenant

### PR DESCRIPTION
**Pull request by nathanfranklin**
## Overview: ##

Add script to migrate project systems to v3 tenant (based on https://github.com/TACC/Core-Portal/blob/978706f5660ca262ce333e641ae72942df439dfd/server/portal/apps/projects/workspace_operations/shared_workspace_migration.py)

Note that: 
* this has been run once already so project systems are on `designsafe.tapis.io`
* the target is `main` but I could re-work things that this PR goes into `Tapis-v3-redesign` if that is preferred. 

## PR Status: ##

* [X] Ready.

## Related Jira tickets: ##

* [DES-2707](https://tacc-main.atlassian.net/browse/DES-2707)

## Testing Steps: ##
1. make build
2. make start
3. docker exec -it des_django ./manage.py migrate_projects_tapis_systems_from_v2_to_v3 --help
4. docker exec -it des_django ./manage.py migrate_projects_tapis_systems_from_v2_to_v3 --dry-run
5.  You could create a new DS project with members and then run `docker exec -it des_django ./manage.py`.  It would go through all projects but skip ones that it had already done and any new project(s).  This might take a while. 
( or `docker exec -it des_django ./manage.py migrate_projects_tapis_systems_from_v2_to_v3 --update-existing` but would take 1+ hours)

## Notes: ##

I ran this already and all project systems are on the tapis v3 tenant.

I will reach out to discuss the following on slack but wanted to note in on the PR:
-  I also checked roles while doing this on the systems.  117 systems where we can't get the roles. Haven't tracked what is going on there.
- There are 150 projects where are is no pi. some of these have no pi, copi, teamMembers but then there are some users for the roles.

TODO 

- [x] share project systems with anyone who also has a role on v2. (from March 15 slack discussions)

### CODE REVIEW

File: designsafe/apps/api/projects_v2/management/commands/migrate_projects_tapis_systems_from_v2_to_v3.py [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+        if dry_run:
+            logger.info("Running in dry-run mode. No changes will be made.")
+
+        total_number_projects = len(list(iterate_entities()))
+
+        for i, project in enumerate(iterate_entities()):
+            uuid = project["uuid"]
+            project_id = project["value"]["projectId"]
+            system = f"project-{uuid}"
+            project_root_dir = f"/corral-repl/projects/NHERI/projects/{uuid}"
```
Comment (nathanfranklin) [inline designsafe/apps/api/projects_v2/management/commands/migrate_projects_tapis_systems_from_v2_to_v3.py:157]: @jarosenb .  adding note from slack: we need custom handling for PRJ-1305 which is the special project used to manage data for community data needs to be handled.

**Review by rstijerina — APPROVED** [review]

**Comment by rstijerina** [comment]
Should this point to the feat/Tapis-v3 branch?

**Comment by nathanfranklin** [comment]
> Should this point to the feat/Tapis-v3 branch?

@rstijerina , I think the idea was to run it on the prod machine prior to V3 deployment.

**Comment by nathanfranklin** [comment]
> > Should this point to the feat/Tapis-v3 branch?
> 
> @rstijerina , I think the idea was to run it on the prod machine prior to V3 deployment.

now targeting  feat/Tapis-v3 branch to run post deployment

**Pull request merged**